### PR TITLE
Set Custom Policies for Roles and Permissions

### PR DESCRIPTION
**Pull request by Reflow1319**
I think this is needed so you can set a permission to only allow a set of people to manage roles and permission in the nova dashboard

### CODE REVIEW

**Comment by Reflow1319** [comment]
Can you review my pull request ?

File: src/ToolServiceProvider.php [hunk 1]
```diff
@@ -26,8 +26,8 @@ public function boot()
             $this->routes();
         });
 
-        Gate::policy(config('permission.models.permission'), PermissionPolicy::class);
-        Gate::policy(config('permission.models.role'), RolePolicy::class);
+        Gate::policy(config('permission.models.permission'), config('permission.policy.permission'));
```
Comment (vyuldashev) [inline src/ToolServiceProvider.php]: Seems like a not good idea to inject custom properties in Spatie's configuration file. I would suggest this approach:

```
 \Vyuldashev\NovaPermission\NovaPermissionTool::make()
            ->rolePolicy(RolePolicy::class)
            ->permissionPolicy(PermissionPolicy::class),
```

`Vyuldashev\NovaPermission\RolePolicy` and `Vyuldashev\NovaPermission\PermissionPolicy` should be used by default.

**Comment by Reflow1319** [comment]
@vyuldashev i followed you suggestion and have update the pull request. let me know if this works for you now. hope you can merge this in

**Comment by bilfeldt** [comment]
For any future reference, if anyone had set up mapping of models to policies in the `$policies` parameter of `app/Providers/AuthServiceProvider.php` instead of the custom part of the Spatie configuration (as `config('permission.models.permission')` and `config('permission.models.role') `) file then it will not work after this PR (version v2.6.0).

# Before:
Registring the tool in `app/Providers/NovaServiceProvider.php@tools` like so:
```php
public function tools()
{
    return [
        \Vyuldashev\NovaPermission\NovaPermissionTool::make(),
    ];
}
```

And declaring policies in the `$policies` parameter of `app/Providers/AuthServiceProvider.php` like so:
```php
protected $policies = [
    \Spatie\Permission\Models\Permission::class => \App\Policies\PermissionPolicy::class,
    \Spatie\Permission\Models\Role::class => \App\Policies\RolePolicy::class,
];
```

No longer works! Instead, use the approach below:

# After
Simply remove the registration of the model-policy mapping in the `$policies` parameter of `app/Providers/AuthServiceProvider.php` and add the policies to the tool registration like the readme now explains (in `app/Providers/NovaServiceProvider.php@tools`):

```php
public function tools()
{
    return [
        \Vyuldashev\NovaPermission\NovaPermissionTool::make()
            ->rolePolicy(\App\Policies\RolePolicy::class)
            ->permissionPolicy(\App\Policies\PermissionPolicy::class),
    ];
}
```

**Pull request merged**